src/identities/masterpubkey.ts

```diff
@@ -0,0 +1,342 @@
+}
+
+// util function that parse a derivation path and return the index
+function getIndex(addrExtended: AddressInterfaceExtended) {
+  const derivationPathSplitted = addrExtended.derivationPath.split('/');
+  const index: number = parseInt(
+    derivationPathSplitted[derivationPathSplitted.length - 1]
+  );
+
+  return index;
+}
+
+export class MasterPublicKey extends Identity implements IdentityInterface {
+  static INITIAL_BASE_PATH: string = "m/84'/0'/0'";
+  static INITIAL_INDEX: number = 0;
+
+  private baseDerivationPath: string = MasterPublicKey.INITIAL_BASE_PATH;
+  private index: number = MasterPublicKey.INITIAL_INDEX;
+  private changeIndex: number = MasterPublicKey.INITIAL_INDEX;
+  private scriptToAddressCache: BufferMap<
+    AddressInterfaceExtended
+  > = new BufferMap();
+
+  readonly masterPublicKeyNode: BIP32Interface;
+  readonly masterBlindingKeyNode: Slip77Interface;
+
+  readonly isRestored: Promise<boolean>;
+
+  constructor(args: IdentityOpts) {
+    super(args);
+
+    // check the identity type
+    if (args.type !== IdentityType.MasterPublicKey) {
+      throw new Error(
+        'The identity arguments have not the MasterPublicKey type.'
+      );
+    }
+    // check the arguments
+    if (!instanceOfMasterPublicKeyOptsValue(args.value)) {
+      throw new Error(
+        'The value of IdentityOpts is not valid for MasterPublicKey Identity.'
+      );
+    }
+    // validate xpub
+    if (!isValidXpub(args.value.masterPublicKey, this.network)) {
+      throw new Error('Master public key is not valid');
+    }
+    // validate master blinding key
+    if (!isValidBlindPub(args.value.masterBlindingKey)) {
+      throw new Error('Master blinding key is not valid');
+    }
+
+    this.masterPublicKeyNode = fromBase58(
+      toXpub(args.value.masterPublicKey),
+      this.network
+    );
+    this.masterBlindingKeyNode = fromMasterBlindingKey(
+      args.value.masterBlindingKey
+    );
+
+    this.isRestored = new Promise(() => true);
+    if (args.initializeFromRestorer) {
+      // restore from restorer
+      this.isRestored = this.restore().catch((reason: any) => {
+        throw new Error(`Error during restoration step: ${reason}`);
+      });
+    }
+  }
+
+  private getCurrentDerivationPath(isChange: boolean): string {
+    const changeValue: number = isChange ? 1 : 0;
+    return `${this.baseDerivationPath}/${changeValue}`;
+  }
+
+  /**
+   * return the next public key derivated from the baseNode.
+   * increment the private member index +1.
+   */
+  private derivePublicKeyWithIndex(isChange: boolean, index: number): Buffer {
+    const baseNode = this.masterPublicKeyNode.derivePath(
+      this.getCurrentDerivationPath(isChange)
+    );
+    const child: BIP32Interface = baseNode.deriveHardened(index);
+
+    return child.publicKey;
+  }
+
+  /**
+   * Derives the script given as parameter to a keypair (SLIP77).
+   * @param scriptPubKey script to derive.
+   */
+  private getBlindingKeyPair(
```

Comment:
I would call this also `derive...`
```suggestion
  private deriveBlindingKeyPair(
```

Reply:
Noted, will make the change in a dedicated PR, cause `Mnemonic` also implement this.